openedx/core/djangoapps/oauth_dispatch/dot_overrides/validators.py

```diff
@@ -29,6 +31,25 @@ def on_access_token_presave(sender, instance, *args, **kwargs):  # pylint: disab
         RestrictedApplication.set_access_token_as_expired(instance)
 
 
+# TODO: Remove Django 1.11 upgrade shim
+# SHIM: Allow users that are inactive to still authenticate while keeping rate-limiting functionality.
+if django.VERSION < (1, 10):
+    from django.contrib.auth.backends import ModelBackend as UserModelBackend
+else:
+    # Django 1.10+ disallows inactive users from authenticating unless using the backend below.
+    from django.contrib.auth.backends import AllowAllUsersModelBackend as UserModelBackend
+
+
+class EdxRateLimitedAllowAllUsersModelBackend(RateLimitMixin, UserModelBackend):
+    """
+    Authentication backend needed to incorporate rate limiting of logins - but also
+    enabling users with is_active of False in the Django auth_user model to still authenticate,
+    which is necessary for mobule users who have not activated their accounts.
```

Comment:
Argh.  Kind of renders the whole `is_active` field rather pointless...

Reply:
Oh boo, did this turn out to be necessary after all? I thought we had worked around a need for one of these?

Reply:
@bmedx I think that this change might not be needed anymore after the CMS changes. Didn't you make a more targeted change to fix this specific condition there? This change represents my ham-handed attempt to fix it.